jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/JibPlugin.java

```diff
@@ -107,21 +106,29 @@ public void apply(Project project) {
     project.afterEvaluate(
         projectAfterEvaluation -> {
           try {
-            // Find project dependencies
-            List<Task> computedDependencies =
-                getProjectDependencyAssembleTasks(projectAfterEvaluation);
             // Has all tasks depend on the 'classes' task.
             Task classesTask = projectAfterEvaluation.getTasks().getByPath("classes");
-            computedDependencies.add(classesTask);
-
-            // dependsOn takes an Object... type
-            Object[] dependenciesArray = computedDependencies.toArray();
-
-            buildImageTask.dependsOn(dependenciesArray);
-            dockerContextTask.dependsOn(dependenciesArray);
-            buildDockerTask.dependsOn(dependenciesArray);
-            buildTarTask.dependsOn(dependenciesArray);
+            buildImageTask.dependsOn(classesTask);
+            dockerContextTask.dependsOn(classesTask);
+            buildDockerTask.dependsOn(classesTask);
+            buildTarTask.dependsOn(classesTask);
 
+            // Find project dependencies and add a dependency to their assemble task. We make sure
+            // to only add the dependency after BasePlugin is evaluated as otherwise the assemble
+            // task may not be available yet.
+            List<Project> computedDependencies = getProjectDependencies(projectAfterEvaluation);
+            for (Project p : computedDependencies) {
```

Comment:
tiny style nit: In this codebase, we tend to avoid single-letter variable names - `dependencyProject` would be a more descriptive name.